specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+        /// Defaults to false.
+        /// </summary>
+        Boolean ValidateFiles{ get; set; };
+    };
+
+    /// Options for creating a bundle
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass BundleOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into Bundle
+        /// </summary>
+        BundleOptions();
+
+        /// <summary>
+        /// Output path for the full bundled file.
+        /// </summary>
+        String BundleFilePath{ get; set; };
+
+        /// <summary>
+        /// Overwrite BundleFilePath if they already exist.
+        /// Defaults to true.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+
+        /// <summary>
+        /// Create as a flat bundle. Package locations will be stored as external path references
+        /// rather than being stored inside the msixbundle file itself. 
+        /// Defaults to false.
+        /// </summary>
+        Boolean FlatBundle{ get; set; };
+
+        /// <summary>
+        /// Optional. Sets the version in the AppxBundleManifest.xml
+        /// If not set the version is created based on the current time.
+        /// </summary>
+        Windows.ApplicationModel.PackageVersion BundleVersion{ get; set; };
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnpackOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnpackOptions();
+
+        /// <summary>
+        /// The output folder to unpack the package into.
+        /// </summary>
+        String UnpackedPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnpackedPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnbundleOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnbundleOptions();
+
+        /// <summary>
+        /// The output folder to unpack the bundle package into.
+        /// </summary>
+        String UnbundledPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnbundledPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateMountableImageOptions
```

Comment:
Missing property?

```
enum MountableImageFormat
{
    Auto,
    CIM,
    VHD,
    VHDX,
}
MoutableImageFormat Format;
```

see earlier comment

Reply:
Still not sure of benefit of this vs using file extensions. Not opposed, just not sure who wouldnt use Auto.